### PR TITLE
Logging improvements

### PR DESCRIPTION
**Pull request by sschuberth**

### CODE REVIEW

**Comment by ge0ffrey** [comment]
@ronmamo Can straitforward PR's like this one be merged?

**Comment by ronmamo** [comment]
.

**Comment by ronmamo** [comment]
and @ge0ffrey if you have time and will, would be nice to assist in merging pr's once in a while, wdyt?

**Comment by ge0ffrey** [comment]
@ronmamo Yes, sounds good.

File: src/main/java/org/reflections/Reflections.java [hunk 1]
```diff
@@ -180,8 +180,8 @@ protected void scan() {
             return;
         }
 
-        if (log != null && log.isDebugEnabled()) {
-            log.debug("going to scan these urls:\n" + Joiner.on("\n").join(configuration.getUrls()));
+        if (log != null) {
+            log.debug("going to scan these urls:\n{}", Joiner.on("\n").join(configuration.getUrls()));
```
Comment (ge0ffrey) [inline src/main/java/org/reflections/Reflections.java]: Just noticed that this one is actually a performance degradation:
The Joiner code does get executed (it's not a lambda here) when debug is disabled, while previously it did not. It's only the string concatenation that doesn't happen when debug is disabled due to the {} usage.

Comment (sschuberth) [reply src/main/java/org/reflections/Reflections.java]: Ah, true, sorry for that. Should I file a PR to revert this hunk?

Comment (ge0ffrey) [reply src/main/java/org/reflections/Reflections.java]: I think only the removal of the "&& log.isDebugEnabled()" etc from the if's should be enough.
The use of lamdba isn't an option yet due to the below java 8 compatibility needs.

**Pull request merged**